src/scripts/main.js

```diff
@@ -1 +1,68 @@
 'use strict';
+
+const body = document.body;
+
+const firstPromise = new Promise((resolve, reject) => {
+  body.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  body.addEventListener('mousedown', e => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  body.addEventListener('mousedown', e => {
+    if (e.button === 0) {
```

Comment:
Your solution is correct but I believe there's better way. Read about this event
https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event